app/models/poll/question.rb

```diff
@@ -69,23 +74,51 @@ def max_votes
     end
   end
 
-  def find_or_initialize_user_answer(user, option_id)
-    option = question_options.find(option_id)
+  def find_or_initialize_user_answer(user, option_id: nil, answer_text: nil)
+    answer = answers.find_or_initialize_by(find_by_attributes(user, option_id))
+
+    if accepts_options?
+      option = question_options.find(option_id)
+      answer.option = option
+      answer.answer = option.title
+    else
+      answer.answer = answer_text
+    end
 
-    answer = answers.find_or_initialize_by(find_by_attributes(user, option))
-    answer.option = option
-    answer.answer = option.title
     answer
   end
 
+  def open_ended_valid_answers_count
+    answers.count
+  end
+
+  def open_ended_blank_answers_count
+    poll.voters.count - open_ended_valid_answers_count
+  end
+
+  def open_ended_answered_percentage
+    return 0.0 if open_ended_total_answers.zero?
+
+    (open_ended_valid_answers_count * 100.0) / open_ended_total_answers
+  end
+
+  def open_ended_unanswered_percentage
```

Comment:
Since we've renamed the other method:

```suggestion
  def open_ended_blank_percentage
```